GlobalstatsIO.cs

```diff
@@ -5,318 +5,315 @@
-		}
-
-		GlobalstatsIO_Leaderboard leaderboard = JsonUtility.FromJson<GlobalstatsIO_Leaderboard>(www.text);
-
-		return leaderboard;
-	}
-	
-}
+    [Serializable]
+    public class StatisticValues
+    {
+        public string key = null;
+        public string value = "0";
+        public string sorting = null;
+        public string rank = "0";
+        public string value_change = "0";
+        public string rank_change = "0";
+    }
+
+    [Serializable]
+    public class LinkData
+    {
+        public string url = null;
+        public string pin = null;
+    }
+
+    [Serializable]
+    public class LeaderboardValue
+    {
+        public string name = null;
+        public string user_profile = null;
+        public string user_icon = null;
+        public string rank = "0";
+        public string value = "0";
+    }
+
+    [Serializable]
+    public class Leaderboard
+    {
+        public LeaderboardValue[] data;
+    }
+
+    public class GlobalstatsIOClient
+    {
+        [HideInInspector]
+        public static string api_id = "";
+
+        [HideInInspector]
+        public static string api_secret = "";
+
+        [Serializable]
+        private class AccessToken
+        {
+            public string access_token = null;
+            public string token_type = null;
+            public string expires_in = null;
+            public int created_at = (int)(DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1))).TotalSeconds;
+
+            public bool IsValid()
+            {
+                //Check if still valid, allow a 2 minute grace period
+                return (created_at + int.Parse(expires_in) - 120) > (int)DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1)).TotalSeconds;
+            }
+        }
+
+        [Serializable]
+        private class StatisticResponse
+        {
+            public string name = null;
+            public string _id = null;
+
+            [SerializeField]
+            public List<StatisticValues> values = null;
+        }
+
+        private AccessToken api_access_token = null;
+
+        private List<StatisticValues> statistic_values = new List<StatisticValues>();
+
+        [HideInInspector]
+        public string statistic_id = "";
+
+        [HideInInspector]
+        public string user_name = "";
+
+        [HideInInspector]
+        public LinkData link_data = null;
+
+        private IEnumerator GetAccessToken()
+        {
+            if (api_id == "" || api_secret == "")
+            {
+                throw new Exception("Credentials missing");
+            }
+
+            string url = "https://api.globalstats.io/oauth/access_token";
+
+            WWWForm form = new WWWForm();
+            form.AddField("grant_type", "client_credentials");
+            form.AddField("scope", "endpoint_client");
+            form.AddField("client_id", api_id);
+            form.AddField("client_secret", api_secret);
+
+            UnityWebRequest www = UnityWebRequest.Post(url, form);
```

Comment:
`UnityWebRequest` implements the dispose pattern.
You should make use of the `using` directive to cleanup the resources.
See: https://docs.unity3d.com/ScriptReference/Networking.UnityWebRequest.Post.html